### DISCUSSION TITLE
[Feature] Refactored search

### DISCUSSION THREAD

**Discussion by alex-phillips** [immich-app/immich#3156] [General]
### The feature

Getting the discussion started for a refactor to the search functionality. Several items discussed:

- Possibly expanding the syntax of the search to include more manual / specific identifiers. Currently there is `m:`, but what about something like `tag:tag one, tag two`? Could also include `people:` as well as other asset properties.
- "Specific" searching vs fuzzy searching. Perhaps this should be more of a 'filter' and then the search itself is for fuzzy searching. Filters would be perfect for a set of items we can pull from the database such as camera make/model.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
An example of the sort of filter UI that is intended: 
![image](https://github.com/immich-app/immich/assets/11602424/88fa7d99-eb89-4ced-88ba-32c41cc69bb0)

In the free-text search field, it would be nice to have typeahead support that suggests fields to filter by.

**Comment by Cotywio** [comment]
Knowing nothing about how search works on the back end, is there any reason why we can't combine the results from object recognition and people/exif info? That way I can search for something like, "John Smith at the beach on June 19th" or something similar (preferably without having to type "m:", hard to push that to regular users). If this is already something that can be done or the wrong thread for this I apologize.

**Reply by bo0tzz** [reply]
Doing it that seamlessly would be pretty tricky, but it should certainly be possible to filter by "John Smith" and "June 19th", and then do a CLIP search within those filters for "at the beach"

**Comment by Moelf** [comment]
I think it's pretty reasonable to be able to combine different search, `tag:` and `people:` are not exposed right now.

I understand sequential refining is hard (e.g. `clip: x`, then `clip: blah`), but the above should be doable?

**Reply by tsemachh** [reply]
I would recommend considering using the [typesense aloglia ui adapter](https://typesense.org/docs/guide/search-ui-components.html#using-instantsearch-js)
Mainly integrate some [facets ](https://www.algolia.com/doc/guides/managing-results/refine-results/faceting/#whats-a-facet)like album , people tag etc.
There's SvelteKit [adaption ](https://github.com/aymeric-giraudet/svelte-algolia-instantsearch)of algolia ui , still need to match it all together , but my experience is really good with this UI.

**Comment by Th0mC** [comment]
Is it, or will be, possible to search for photos with 2 persons in it?

**Comment by davidXire** [comment]
The idea behind this feature, is it also to be able to search with data inside exif (like tag put with lightroom) ?
